src/coordinator/api/v1/handler/prometheus/remote/write.go

```diff
@@ -104,11 +121,70 @@ func (h *PromWriteHandler) parseRequest(r *http.Request) (*prompb.WriteRequest,
 }
 
 func (h *PromWriteHandler) write(ctx context.Context, r *prompb.WriteRequest) error {
-	requests := make([]execution.Request, len(r.Timeseries))
-	for idx, t := range r.Timeseries {
-		requests[idx] = newLocalWriteRequest(storage.PromWriteTSToM3(t), h.store)
+	var (
+		wg                   sync.WaitGroup
+		writeUnaggregatedErr error
+		writeAggregatedErr   error
+	)
+	if h.downsampler != nil {
```

Comment:
this block can be another function ? Similarly, h.store != nil block can be another function

Reply:
Yup, good call. Done.